pathd/path_nb_config.c

```diff
@@ -0,0 +1,731 @@
+
+	segment = nb_running_get_entry(args->dnode, NULL, true);
+	sid_value = yang_dnode_get_uint32(args->dnode, NULL);
+	segment->sid_value = sid_value;
+	SET_FLAG(segment->segment_list->flags, F_SEGMENT_LIST_MODIFIED);
+
+	return NB_OK;
+}
+
+int pathd_srte_segment_list_segment_nai_destroy(struct nb_cb_destroy_args *args)
+{
+	struct srte_segment_entry *segment;
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	segment = nb_running_get_entry(args->dnode, NULL, true);
+	segment->nai_type = SRTE_SEGMENT_NAI_TYPE_NONE;
+	segment->nai_local_addr.ipa_type = IPADDR_NONE;
+	segment->nai_local_iface = 0;
+	segment->nai_remote_addr.ipa_type = IPADDR_NONE;
+	segment->nai_remote_iface = 0;
+
+	return NB_OK;
+}
+
+void pathd_srte_segment_list_segment_nai_apply_finish(
+	struct nb_cb_apply_finish_args *args)
+{
+	struct srte_segment_entry *segment;
+	enum srte_segment_nai_type type;
+	struct ipaddr local_addr, remote_addr;
+	uint32_t local_iface = 0, remote_iface = 0;
+
+	segment = nb_running_get_entry(args->dnode, NULL, true);
+	type = yang_dnode_get_enum(args->dnode, "./type");
+
+	yang_dnode_get_ip(&local_addr, args->dnode, "./local-address");
+
+	switch (type) {
+	case SRTE_SEGMENT_NAI_TYPE_IPV4_NODE:
+	case SRTE_SEGMENT_NAI_TYPE_IPV6_NODE:
+		break;
+	case SRTE_SEGMENT_NAI_TYPE_IPV4_ADJACENCY:
+	case SRTE_SEGMENT_NAI_TYPE_IPV6_ADJACENCY:
+		yang_dnode_get_ip(&remote_addr, args->dnode,
+				  "./remote-address");
+		break;
+	case SRTE_SEGMENT_NAI_TYPE_IPV4_UNNUMBERED_ADJACENCY:
+		yang_dnode_get_ip(&remote_addr, args->dnode,
+				  "./remote-address");
+		local_iface =
+			yang_dnode_get_uint32(args->dnode, "./local-interface");
+		remote_iface = yang_dnode_get_uint32(args->dnode,
+						     "./remote-interface");
+		break;
+	default:
+		break;
+	}
+
+	srte_segment_entry_set_nai(segment, type, &local_addr, local_iface,
+				   &remote_addr, remote_iface);
+}
+
+/*
+ * XPath: /frr-pathd:pathd/srte/policy
+ */
+int pathd_srte_policy_create(struct nb_cb_create_args *args)
+{
+	struct srte_policy *policy;
+	uint32_t color;
+	struct ipaddr endpoint;
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	color = yang_dnode_get_uint32(args->dnode, "./color");
+	yang_dnode_get_ip(&endpoint, args->dnode, "./endpoint");
+	policy = srte_policy_add(color, &endpoint);
+
+	nb_running_set_entry(args->dnode, policy);
+	SET_FLAG(policy->flags, F_POLICY_NEW);
+
+	return NB_OK;
+}
+
+int pathd_srte_policy_destroy(struct nb_cb_destroy_args *args)
+{
+	struct srte_policy *policy;
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	policy = nb_running_unset_entry(args->dnode);
+	SET_FLAG(policy->flags, F_POLICY_DELETED);
+
+	return NB_OK;
+}
+
+/*
+ * XPath: /frr-pathd:pathd/srte/policy/name
+ */
+int pathd_srte_policy_name_modify(struct nb_cb_modify_args *args)
+{
+	struct srte_policy *policy;
+	const char *name;
+
+	if (args->event != NB_EV_APPLY && args->event != NB_EV_VALIDATE)
+		return NB_OK;
+
+	policy = nb_running_get_entry(args->dnode, NULL, true);
```

Comment:
should `policy` be checked before using?

Reply:
Same as for the segment list.